Vates/VatesAPI/test/vtkDataSetFactoryTest.h

```diff
@@ -53,28 +53,35 @@ class vtkDataSetFactoryTest : public CxxTest::TestSuite
   };
 
 public:
-
-  void testSetSuccessor()
-  {
+  void testSetSuccessor() {
     MockvtkDataSetFactory factory;
-    MockvtkDataSetFactory* pSuccessor = new MockvtkDataSetFactory;
-    
-    EXPECT_CALL(factory, getFactoryTypeName()).WillOnce(testing::Return("TypeA")); 
-    EXPECT_CALL(*pSuccessor, getFactoryTypeName()).WillOnce(testing::Return("TypeB")); //Different type name, so setting the successor should work.
-    factory.SetSuccessor(pSuccessor);
+    auto successor = Mantid::Kernel::make_unique<MockvtkDataSetFactory>();
+    MockvtkDataSetFactory *pSuccessor = successor.get();
+
+    EXPECT_CALL(factory, getFactoryTypeName())
+        .WillOnce(testing::Return("TypeA"));
+    EXPECT_CALL(*successor, getFactoryTypeName())
+        .WillOnce(testing::Return("TypeB")); // Different type name, so setting
+                                             // the successor should work.
+    factory.SetSuccessor(std::move(successor));
 
     TSM_ASSERT("Successor should have been set", factory.hasSuccessor());
     TS_ASSERT(testing::Mock::VerifyAndClearExpectations(&factory));
     TS_ASSERT(testing::Mock::VerifyAndClearExpectations(pSuccessor));
   }
 
-  void testSetSuccessorThrows()
-  {
+  void testSetSuccessorThrows() {
     MockvtkDataSetFactory factory;
-    MockvtkDataSetFactory* pSuccessor = new MockvtkDataSetFactory;
-    EXPECT_CALL(factory, getFactoryTypeName()).WillOnce(testing::Return("TypeA")); 
-    EXPECT_CALL(*pSuccessor, getFactoryTypeName()).WillOnce(testing::Return("TypeA")); //Same type name. should NOT work.
-    TSM_ASSERT_THROWS("By default, should throw when successor type is the same as the container.", factory.SetSuccessor(pSuccessor), std::runtime_error);
+    auto successor = Mantid::Kernel::make_unique<MockvtkDataSetFactory>();
+    MockvtkDataSetFactory *pSuccessor = successor.get();
```

Comment:
`pSuccessor` seems to be used only once? Defining it was more confusing than helpful for me.